From 34382b30ae61d6c77276f145a9619ade274ee0b2 Mon Sep 17 00:00:00 2001
From: raosrini <srinidhi.rao@intel.com>
Date: Mon, 27 Apr 2015 11:35:50 +0530
Subject: [PATCH] EM/FG: invoke get_long_avg_ocv function ptr after NULL check

The function pointer from the coulomb counter driver "get_long_avg_ocv"
should be invoked under an if check for NULL pointer so that NULL
pointer de-referencing can be prevented.
This patch fixes the Klock-Work issue 81439

Change-Id: I821f4dd030204ca88dee7083fd34ef5b015d6a56
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1486
Signed-off-by: raosrini <srinidhi.rao@intel.com>
---
 drivers/power/intel_fuel_gauge.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/power/intel_fuel_gauge.c b/drivers/power/intel_fuel_gauge.c
index 9a45e7c1653e..9a4e0421991e 100644
--- a/drivers/power/intel_fuel_gauge.c
+++ b/drivers/power/intel_fuel_gauge.c
@@ -283,7 +283,7 @@ static void intel_fg_worker(struct work_struct *work)
 			dev_err(fg_info->dev, "Error while reading UP CC");
 	}
 	if (fg_info->input->get_long_avg_ocv) {
-		ret = fg_info->input->get_long_avg(&ip.long_avg_at_ocv);
+		ret = fg_info->input->get_long_avg_ocv(&ip.long_avg_at_ocv);
 		if (ret)
 			dev_err(fg_info->dev, "Error while reading UP CC");
 	}
-- 
1.9.1

